### PR TITLE
[All] Update for contract inheritance

### LINKED ISSUE #124: [All] Update contract functions as `virtual`
Update contract functions as virtual for inheritance.
- Make functions as virtual functions
- Check & Update local state variables from `private` to `internal` to be used in extended contracts

### PR DESCRIPTION
**Pull request by ITStar10**
- Update contract functions as `virtual`
- Update contract state variables from `private` to `internal`
- Added __gap in the state variables for later use

### CODE REVIEW

File: VDA-DID-Registry/contracts/VeridaDIDRegistryV2.sol [hunk 1]
```diff
@@ -1,232 +1,11 @@
...
-    address[] memory ret = new address[](count);
-
-    for (uint i; i < count;) {
-      ret[i] = _registeredDIDs.at(startIndex + i);
-      unchecked { ++i; }
-    }
-
-    return ret;
+contract VeridaDIDRegistryV2 is VeridaDIDRegistry {
+  function getVerison() external pure virtual returns(string memory) {
```
Comment (tahpot) [inline VDA-DID-Registry/contracts/VeridaDIDRegistryV2.sol]: Typo

Comment (tahpot) [reply VDA-DID-Registry/contracts/VeridaDIDRegistryV2.sol]: Add a unit test for this.

**Review by tahpot — CHANGES_REQUESTED** [review]

File: VDA-DID-Registry/contracts/VeridaDIDRegistryV2.sol (hunk 1 above)
Comment (ITStar10) [reply VDA-DID-Registry/contracts/VeridaDIDRegistryV2.sol]: This `VeridaDIDRegistryV2` contract was created for upgrade test in the testnet.
What unit tests do you need?

Comment (tahpot) [reply VDA-DID-Registry/contracts/VeridaDIDRegistryV2.sol]: Just a test that `getVersion()` returns a value.

That would pickup the typo (`getVerison()`)

**Comment by tahpot** [comment]
Note: `__gap` was removed from `VDAVerificationBase` as it would break backwards compatibility with previously deployed contracts on testnet.

See https://github.com/verida/blockchain-contracts/pull/125/commits/e424e6218716fe1bfa3e6592bde7648c77b8871b

File: VDA-DID-Registry/contracts/VeridaDIDRegistryV2.sol (hunk 1 above)
Comment (ITStar10) [reply VDA-DID-Registry/contracts/VeridaDIDRegistryV2.sol]: Updated

**Review by tahpot — APPROVED** [review]

**Pull request merged**